Mathlib/LinearAlgebra/BilinearMap.lean

```diff
@@ -208,6 +208,45 @@ theorem domRestrict₁₂_apply (f : M →ₛₗ[ρ₁₂] N →ₛₗ[σ₁₂]
     (x : p) (y : q) : f.domRestrict₁₂ p q x y = f x y := rfl
 #align linear_map.dom_restrict₁₂_apply LinearMap.domRestrict₁₂_apply
 
+section restrictScalars
+
+variable (R' S' : Type*)
+
+variable [Semiring R'] [Semiring S'] [Module R' M] [Module S' N] [Module R' Pₗ] [Module S' Pₗ]
+
+variable [SMulCommClass S' R' Pₗ]
+
+variable [SMul S' S] [IsScalarTower S' S N] [IsScalarTower S' S Pₗ]
+
+variable [SMul R' R] [IsScalarTower R' R M] [IsScalarTower R' R Pₗ]
+
+/-- If `B : M → N → Pₗ` is `R`-`S` bilinear and `R'` and `S'` are compatible scalar multiplications,
+then the restriction of scalars is a `R'`-`S'` bilinear map.-/
+def restrictScalars₁₂ (B : M →ₗ[R] N →ₗ[S] Pₗ) : M →ₗ[R'] N →ₗ[S'] Pₗ :=
+  LinearMap.mk₂' R' S'
+    (fun x y ↦ B x y)
+    (fun _ _ _ ↦ B.map_add₂ _ _ _)
+    (fun r' m n ↦ by
+      dsimp only
+      rw [← smul_one_smul R r' m, map_smul₂, smul_one_smul])
+    (fun x ↦ map_add (B x))
+    (fun _ x ↦ (B x).map_smul_of_tower _)
+
+@[simp]
+theorem restrictScalars₁₂_apply (B : M →ₗ[R] N →ₗ[S] Pₗ) (x : M) (y : N) :
+    B.restrictScalars₁₂ R' S' x y = B x y := rfl
```

Comment:
`@[simps]` can autogenerate this but it will be called `_apply_apply`. Is that the reason for manually defining the lemma?

Reply:
I think I'd be inclined to keep the `_apply_apply` name anyway